bridge/github/config.go

```diff
@@ -232,8 +285,25 @@ func promptURL() (string, string, error) {
 	}
 }
 
+func checkGithubURL(url string) bool {
+
+	re, err := regexp.Compile(`^[a-z]*\W*github\.com[a-z]*.*$`)
```

Comment:
Would it make much of a difference if we do this.Rather keeping it via multiple calls would still be sane IMO.

Reply:
The problem with this is that: If there is an error in the regex expression, the only way to detect it would be to try it manually (runtime). Using `MustCompile` (outside of the function) will prevent that by checking the expressions in the compile time.

Compiling regexes is costing a lot also, you can check the difference your self with this test file.
```golang
package git

import (
	"regexp"
	"testing"
)

var Sink bool

func BenchmarkCompileRun(b *testing.B) {
	for i := 0; i < b.N; i++ {
		rx := regexp.MustCompile(`[\w\.+-]+@[\w\.-]+\.[\w\.-]+`)
		Sink = rx.MatchString("123456789 foo@bar.etc")
	}
}

func BenchmarkRun(b *testing.B) {
	rx := regexp.MustCompile(`[\w\.+-]+@[\w\.-]+\.[\w\.-]+`)
	for i := 0; i < b.N; i++ {
		Sink = rx.MatchString("123456789 foo@bar.etc")
	}
}

```

```bash
$ go test -bench=.

goos: darwin
goarch: amd64
pkg: gitlab.com/a-hilaly/blockchain
BenchmarkCompileRun-8             300000              5192 ns/op
BenchmarkRun-8                   2000000               938 ns/op
PASS
ok      gitlab.com/a-hilaly/blockchain   4.458s
```